webrtc/src/peer_connection/mod.rs

```diff
@@ -1370,32 +1359,22 @@ impl RTCPeerConnection {
                                 t.set_mid(mid_value.to_owned()).await?;
                             }
                         } else {
-                            let receiver = Arc::new(RTCRtpReceiver::new(
-                                self.internal.setting_engine.get_receive_mtu(),
-                                kind,
-                                Arc::clone(&self.internal.dtls_transport),
-                                Arc::clone(&self.internal.media_engine),
-                                Arc::clone(&self.interceptor),
-                            ));
-
                             let local_direction =
                                 if direction == RTCRtpTransceiverDirection::Recvonly {
                                     RTCRtpTransceiverDirection::Sendonly
                                 } else {
                                     RTCRtpTransceiverDirection::Recvonly
                                 };
 
-                            let t = RTCRtpTransceiver::new(
-                                Some(receiver),
-                                None,
-                                local_direction,
-                                kind,
-                                vec![],
-                                Arc::clone(&self.internal.media_engine),
-                                Some(Box::new(self.internal.make_negotiation_needed_trigger())),
-                            )
-                            .await;
-
+                            let t = self
```

Comment:
I'm not sure we should use the public API here. The public API runs in a different context from creating a transceiver because we apply an SDP answer

Reply:
> Nice! I left a few comments.
> 
> Did you test this with at least some of the examples to make sure it hasn't broken anything?

Yes, I tested with 
cargo test,
using internal example,

and on a side project.

Maybe I didn't cover every possible case, but nothing broke

Reply:
fixed